numpy/polynomial/_polybase.py

```diff
@@ -70,12 +79,28 @@ class ABCPolyBase(abc.ABC):
     @property
     @abc.abstractmethod
     def domain(self):
-        pass
+        return self._domain
+
+    @domain.setter
+    @abc.abstractmethod
+    def domain(self, domain):
+        [domain] = pu.as_series([domain], trim=False)
+        if len(domain) != 2:
+            raise ValueError("Domain has wrong number of elements.")
+        self._domain = domain
```

Comment:
Could consider:
```suggestion
        if domain is None:
            # remove the instance-specific value to restore the default
            try:
                del self._domain
            except KeyError:
                pass
            return
        [domain] = pu.as_series([domain], trim=False)
        if len(domain) != 2:
            raise ValueError("Domain has wrong number of elements.")
        self._domain = domain
```

Reply:
or
```suggestion
        if domain is None:
            self._domain = type(self)._domain
            return
        [domain] = pu.as_series([domain], trim=False)
        if len(domain) != 2:
            raise ValueError("Domain has wrong number of elements.")
        self._domain = domain
```

Reply:
As I understand it, the main thing that this code does is allow `None` as a valid value for the domain setter. By example:

#### Current behavior of NumPy
```python
>>> p = np.polynomial.Polynomial([1, 2, 3], domain=[-10, 10])
>>> p
Polynomial([1., 2., 3.], domain=[-10.,  10.], window=[-1,  1])
>>> p.domain = None
>>> p.domain

```

#### Behavior with this PR
```python
>>> p = np.polynomial.Polynomial([1, 2, 3], domain=[-10, 10])
>>> p
Polynomial([1., 2., 3.], domain=[-10.,  10.], window=[-1,  1])
>>> p.domain = None
ValueError: Domain has wrong number of elements.
>>> p.domain
array([-10.,  10.])
```

#### Behavior with proposed change
```python
>>> p = np.polynomial.Polynomial([1, 2, 3], domain=[-10, 10])
>>> p 
Polynomial([1., 2., 3.], domain=[-10.,  10.], window=[-1,  1])
>>> p.domain = None
>>> p.domain
array([-1., 1.])
```

Is this correct and the change that you intended? IMO, I prefer the version that errors rather than giving `None` a special meaning that falls back to the default value for the class. This could be achieved via `p.domain = np.polynomial.Polynomial.domain`.